pkg/controller/egressgateway/egressgateway_controller.go

```diff
@@ -0,0 +1,666 @@
+
+	// Get all the Egress Gateway resources available.
+	egws, err := getEgressGateways(ctx, r.client)
+	if err != nil {
+		reqLogger.Error(err, "Error querying for Egress Gateway")
+		r.status.SetDegraded(operatorv1.ResourceReadError, "Error querying for Egress Gateway", err, reqLogger)
+		return reconcile.Result{}, err
+	}
+
+	// If there are no Egress Gateway resources, return.
+	if len(egws) == 0 {
+		r.status.OnCRNotFound()
+		return reconcile.Result{}, nil
+	}
+
+	/* Reconcile is done as follows.
+	1. At the start, assume all the EGW resources need to be reconciled.
+	2. If the request is to a particular EGW resource, find the requested EGW from the list of EGWs.
+	3. If the requested EGW resource is not present, it could have been deleted.
+	   'egws' is now the list of all EGW resources present. Get the cumulative status and update
+	   TigeraStatus object.
+	4. If the requested EGW resource is present, then it is the only resource to reconcile. Remove this
+	   EGW resource from the list and get the status of all the other EGW resources present and update
+	   TigeraStatus accordingly.
+	5. If the request is not to a particular EGW resource, reconcile all the resources.
+	*/
+
+	// egwsToReconcile is the list of Egress Gateway resources that needs to be reconciled.
+	// To start with all EGW resources must be reconciled.
+	egwsToReconcile := egws
+	if request.Namespace != "" {
+		requestedEGW, idx := getRequestedEgressGateway(egws, request)
+		if requestedEGW == nil {
+			// Request object not found, could have been deleted after reconcile request.
+			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
+			reqLogger.Info("EgressGateway object not found")
+			// Since the EGW resource is not found, remove the deployment.
+			r.status.RemoveDeployments(types.NamespacedName{Name: request.Name, Namespace: request.Namespace})
+			// Get the unready EGW. Let's say we have 2 EGW resources red and blue.
+			// Red has already degraded. When the user deletes Red, TigeraStatus should go back to available
+			// as Blue is healthy. If all the EGWs are ready, clear the degraded TigeraStatus.
+			// If at least one of the EGWs is unhealthy, get the degraded msg from the conditions and
+			// update the TigeraStatus.
+			unreadyEGW := getUnreadyEgressGateway(egws)
+			if unreadyEGW != nil {
+				r.status.SetDegraded(operatorv1.ResourceNotReady,
+					fmt.Sprintf("Error reconciling Egress Gateway resource. Name=%s Namespace=%s", unreadyEGW.Name, unreadyEGW.Namespace),
+					fmt.Errorf("%s", getDegradedMsg(unreadyEGW)), reqLogger)
+				return reconcile.Result{}, nil
+			}
+			r.status.ClearDegraded()
+			return reconcile.Result{}, nil
+		}
+		// If the EGW resource is present, reconcile only that resource.
+		// Remove this from the list of EGWs before computing status.
+		egwsToReconcile = []operatorv1.EgressGateway{*requestedEGW}
+		egws = append(egws[:idx], egws[idx+1:]...)
```

Comment:
> // Remove this from the list of EGWs before computing status.

IIUC, I think this is only done because we are trying to handle unready EGW status reporting in this controller. However, once we remove that logic and use the status manager instead, we won't need this line.